### PR TITLE
Stronger channel verification

### PR DESCRIPTION
**Pull request by elGatoMantocko**
### Things I did
- Publish single channels based on id
- Subscribe to single channel in channel page
- Change to a stronger channel page test

### CODE REVIEW

File: client/controllers/public/channel.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Template.channel.onCreated(function() {
+  Meteor.subscribe('oneChannel', FlowRouter.getParam('id'));
+});
+
+Template.channel.helpers({
+  channel: function() {
+    return Channel.getChannel(FlowRouter.getParam('id')).fetch()[0];
```
Comment (elGatoMantocko) [inline client/controllers/public/channel.js]: If anyone can give me a better way to do this, feel free to put it here. I tried a bunch of different stuff.

**Comment by elGatoMantocko** [comment]
@cjsauer This test is a little stronger for channel verification now. I didn't really want to put it in #11 because I was going to be changing the publications a little.

File: client/controllers/public/channel.js (hunk 1 above)
Comment (elGatoMantocko) [inline client/controllers/public/channel.js]: ### Got it!

File: client/controllers/public/channel.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Template.channel.onCreated(function() {
+  Meteor.subscribe('oneChannel', FlowRouter.getParam('id'));
```
Comment (cjsauer) [inline client/controllers/public/channel.js]: This should be wrapped in an autorun block like the [code example here](https://kadira.io/academy/meteor-routing-guide/content/subscriptions-and-data-management/with-blaze). 

That way if the id parameter changes, the subscription will be automatically stopped.

File: collections/Channels.js [hunk 3]
```diff
@@ -62,3 +62,13 @@ Channel.getLatest = function(limit) {
...
+
+/*
+ * Find a single channel by id
+ *
+ * @param {string} id The channel id to fetch
+ * @return {Mongo.Cursor} the resulting Mongo cursor from find()
+ */
+Channel.getChannel = function(id) {
+  return Channel.find({"_id": id});
+}
```
Comment (cjsauer) [inline collections/Channels.js]: It turns out you can just do `Channel.findOne("someIDstring")` so this method is kind of unnecessary. 

The `find` and `findOne` methods are equivalent to the Meteor collection calls. Check out the [findOne doc entry](http://docs.meteor.com/#/full/findone).

File: tests/cucumber/features/channels/channels.feature [hunk 4]
```diff
@@ -31,7 +31,8 @@ Feature: Channels
     And I enter "Galaxy" into the title field
     And I enter "pony+galaxy" into the query field
     And I submit the new channel form
-    Then I should be on the new channel
+    Then channel "title" should display "Galaxy"
+    And channel "query" should display "pony+galaxy"
```
Comment (cjsauer) [inline tests/cucumber/features/channels/channels.feature]: :+1:

**Comment by cjsauer** [comment]
:+1:

**Comment by elGatoMantocko** [comment]
:+1: @hunterbmartin you want to take a look before we merge?

**Comment by marti584** [comment]
:+1:

**Pull request merged**